### PR TITLE
fix: check if exception is defined before proceed

### LINKED ISSUE #248: TypeError: Cannot read property 'conflictInfo' of undefined
## Bug Report

Stack trace:
```
TypeError: Cannot read property 'conflictInfo' of undefined

      at ConflictLink.Object.<anonymous>.ConflictLink.getConflictData (node_modules/offix-offline/src/conflicts/ConflictLink.ts:109:40)
      at ConflictLink.Object.<anonymous>.ConflictLink.conflictHandler (node_modules/offix-offline/src/conflicts/ConflictLink.ts:79:23)
      at Object.error (node_modules/apollo-link-error/src/index.ts:63:29)
      at notifySubscription (node_modules/zen-observable/lib/Observable.js:134:18)
      at onNotify (node_modules/zen-observable/lib/Observable.js:165:3)
      at SubscriptionObserver.error (node_modules/zen-observable/lib/Observable.js:224:7)
      at node_modules/apollo-link-http/src/httpLink.ts:184:20          at runMicrotasks (<anonymous>)
```

most probably my server is returning an error, but the `getConflictData()` is trying to read the `conflictInfo` and it fails, because it is not a conflict error, and at the same time hide the real error 

* **Affected Package(s)**: offix-offline
* **Package version(s)**: 0.8.2
* **Platform (e.g. Android/iOS)**: jest
* **Platform Version**: 24.0.22
* **Cordova Version**: none
* **Node.js / npm versions**: v12.10.0

### PR DESCRIPTION
**Pull request by b1zzu**
### Description

Fixes #248

##### Checklist

- [x] `npm test` passes
- [x] `npm run build` works

### CODE REVIEW

File: packages/offix-offline/src/conflicts/ConflictLink.ts [hunk 1]
```diff
@@ -106,8 +106,10 @@ export class ConflictLink extends ApolloLink {
     if (graphQLErrors) {
       for (const err of graphQLErrors) {
         if (err.extensions) {
-          if (err.extensions.exception.conflictInfo) {
-            return err.extensions.exception.conflictInfo;
+          if (err.extensions.exception) {
+            if (err.extensions.exception.conflictInfo) {
+              return err.extensions.exception.conflictInfo;
+            }
```
Comment (darahayes) [inline packages/offix-offline/src/conflicts/ConflictLink.ts]: could we collapse this down to 1 statement perhaps?

```
if (err.extensions && err.extensions.exeption && err.extensions.exception.conflictInfo)
```

**Comment by darahayes** [comment]
@b1zzu thanks for this. Can you rebase and then we will merge?

**Comment by b1zzu** [comment]
Hey @darahayes I thought that isntat of having that long chain of `if` we chould use typescript chains: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

**Comment by darahayes** [comment]
@b1zzu I thought the very same thing but even though we have the latest version of typescript, I couldn't get it to compile. If you'd like to try that, you are more than welcome!

**Comment by b1zzu** [comment]
@darahayes same problem I was trying it out but it doesn't work, but maybe I find the problem

**Pull request merged**